oxd-server/src/test/java/org/xdi/oxd/server/GetTokensByInvalidCodeTest.java

```diff
@@ -0,0 +1,53 @@
+package org.xdi.oxd.server;
+
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+import org.xdi.oxd.client.ClientInterface;
+import org.xdi.oxd.common.CoreUtils;
+import org.xdi.oxd.common.params.GetTokensByCodeParams;
+import org.xdi.oxd.common.response.GetTokensByCodeResponse;
+import org.xdi.oxd.common.response.RegisterSiteResponse;
+
+import javax.ws.rs.BadRequestException;
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class GetTokensByInvalidCodeTest {
+
+
+    @Parameters({"host", "opHost", "redirectUrl", "userId", "userSecret"})
+    @Test
+    public void test(String host, String opHost, String redirectUrl, String userId, String userSecret) throws IOException {
```

Comment:
Test should be properly named `test` -> `whenInvalidCodeIsUsed_shouldGet400BadRequest`